### DISCUSSION TITLE
Does realtime subscribtion cancels automaticly?

### DISCUSSION THREAD

**Discussion by lixelv** [pocketbase/pocketbase#5428] [Q&A] [answered]
When I'm subscribing to a realtime data, if I won't unsubscribe onDestroy does subscription will cancel automatically?

**Comment by ganigeorgiev** [comment, marked as answer]
It depends both what you are doing and on the used framework (and its execution context).

Usually it is a good idea to cancel the subscription on remove/destroy to avoid leaks (_and eventual hidden reactivity bugs_) in case your component is mounted more than once.

The subscription is automatically cancelled when:

- the client has closed the tab/window
- no new messages have been received for the last ~5mins (it will attempt a reconnect if the client is still alive)
- the server restarts for some reason (it will attempt a reconnect if the client is still alive)

**Comment by jonshipman** [comment]
If you're talking about lifecycles in a client application, it's good practice to manually cancel subscriptions when you're no longer using them.

**Comment by lixelv** [comment]
Thanks for answers. Ok, I've understood. Better to unsubscribe, this is better practice, but also I can just leave subscription, without canceling, and this connection will be closed.
